config.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2015 Brian J. Downs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openweathermap
+
+import (
+	"errors"
+	"os"
+)
+
+// Config will hold default settings to be passed into the
+// "NewCurrent, NewForecast, etc}" functions.
+type Config struct {
+	Mode     string // user choice of JSON or XML
+	Unit     string // measurement for results to be displayed.  F, C, or K
+	Lang     string // should reference a key in the LangCodes map
+	APIKey   string // API Key for connecting to the OWM
+	Username string // Username for posting data
+	Password string // Pasword for posting data
+}
+
+// CheckAPIKeyExists will see if an API key has been set.
+func (c *Config) CheckAPIKeyExists() bool {
+	return len(c.APIKey) > 1
+}
+
+// config will hold default settings to be passed into the OWM requests
+var config = Config{}
```

Comment:
Or Conf... My point is that we don't have to rename the struct to accommodate what a variable might get called. 

> On Oct 23, 2015, at 16:13, Ravi Teja <notifications@github.com> wrote:
> 
> In config.go:
> 
> > +type Config struct {
> > +	Mode     string // user choice of JSON or XML
> > +	Unit     string // measurement for results to be displayed.  F, C, or K
> > +	Lang     string // should reference a key in the LangCodes map
> > +	APIKey   string // API Key for connecting to the OWM
> > +	Username string // Username for posting data
> > +	Password string // Pasword for posting data
> > +}
> > +
> > +// CheckAPIKeyExists will see if an API key has been set.
> > +func (c *Config) CheckAPIKeyExists() bool {
> > +	return len(c.APIKey) > 1
> > +}
> > +
> > +// config will hold default settings to be passed into the OWM requests
> > +var config = Config{}
> if it is called conf it cannot be exported.
> 
> —
> Reply to this email directly or view it on GitHub.
>